### PR TITLE
docs: add demo

### LINKED ISSUE #674: add a quick demo in the readme
**Describe the solution you'd like**

- add a demo (gif or link to asciinema?) that quickly highlights the project in the readme
- add a demo magic script for repros

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Eraser version:
- Kubernetes version: (use `kubectl version`):

### PR DESCRIPTION
**Pull request by ashnamehrotra**
**What this PR does / why we need it**:
Adds demo based on https://github.com/salaxander/eraser-demo to README.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #674

**Special notes for your reviewer**:

### CODE REVIEW

File: demo/fonts/JetBrainsMono-1.0.0.zip
Comment (sozercan) [inline demo/fonts/JetBrainsMono-1.0.0.zip:1]: do we need these fonts?

File: demo/ds.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    matchLabels:
+      app: alpine
+  template:
+    metadata:
+      labels:
+        app: alpine
+    spec:
+      containers:
+      - name: alpine
+        image: docker.io/library/alpine:3.7.3
```
Comment (sozercan) [inline demo/ds.yaml]: nit: newline

File: demo/demo.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+clear
+
+# demo commands
+pei "kubectl get pods"
+sleep 10
+pei "kubectl delete daemonset alpine"
+sleep 5
+pei "kubectl get pods"
+pei "docker exec kind-control-plane ctr -n k8s.io images list | grep alpine"
+pei "helm install -n eraser-system eraser ../manifest_staging/charts/eraser --create-namespace"
```
Comment (sozercan) [inline demo/demo.sh]: can we use the helm repo instead of charts folder

File: demo/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Eraser Demo
+This demo is using [demo magic](https://github.com/paxtonhare/demo-magic) to execute. To run, simply start the `demo.sh` script!
+
+```
+$ ./demo.sh
+```
```
Comment (sozercan) [inline demo/README.md]: can we add the prereqs for this? sounds like kind, docker, kubectl, etc

File: demo/fonts/JetBrainsMono-1.0.0.zip
Comment (ashnamehrotra) [reply demo/fonts/JetBrainsMono-1.0.0.zip:1]: no that was just to conver the gif locally, will remove it

File: demo/demo.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+########################
+# include the magic
+########################
+. demo-magic.sh
+
+# boostrap environment
+pei "kind create cluster"
```
Comment (sozercan) [inline demo/demo.sh]:
```suggestion
pei "kind create cluster --name eraser-demo"
```

File: demo/demo.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+pei "docker exec kind-control-plane ctr -n k8s.io images list | grep alpine"
+pei "helm install -n eraser-system eraser eraser/eraser --create-namespace"
+sleep 5
+pei "kubectl get po -n eraser-system"
+sleep 60
+pei "kubectl get po -n eraser-system"
+pei "docker exec kind-control-plane ctr -n k8s.io images list | grep alpine"
+
+# teardown environment
+kind delete cluster
```
Comment (sozercan) [inline demo/demo.sh]:
```suggestion
kind delete cluster --name eraser-demo
```

so we avoid accidentally deleting someone's kind default cluster

File: demo/demo.sh (hunk 4 above)
Comment (sozercan) [reply demo/demo.sh]: this might change the container name

File: demo/demo.sh [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+sleep 10
+pei "kubectl delete daemonset alpine"
+sleep 5
+pei "kubectl get pods"
+pei "docker exec kind-control-plane ctr -n k8s.io images list | grep alpine"
+pei "helm install -n eraser-system eraser eraser/eraser --create-namespace"
+sleep 5
+pei "kubectl get po -n eraser-system"
+sleep 60
+pei "kubectl get po -n eraser-system"
```
Comment (sozercan) [inline demo/demo.sh:26]: can we wait instead of sleeping here

**Comment by ashnamehrotra** [comment]
going to record the demo again once #675 is merged so the pod names stay consistent

**Review by sozercan — APPROVED** [review]
LGTM

**Pull request merged**